### PR TITLE
Update setup.py to fix the issue that f2py doesn't set '-shared' link option in some linux system

### PR DESCRIPTION
**Pull request by shenmidelin**
The error "start.S:99: undefined reference to `main'"  from building fortran extension in [docker nsls2/debian-with-miniconda](https://hub.docker.com/r/nsls2/debian-with-miniconda/tags/) is due to that f2py doesn't set '-shared' link option in the link stage, so I update the setup.py to explicitly set that option when the system is Linux. 

MANIFEST.in file is also updated to include src/* in distribution package.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -35,8 +35,13 @@
                     if not line.startswith('#')]
 
 # Python interface to call fortran subroutines
+if 'linux' in sys.platform:
+    extra_link_args=['-shared']
```
Comment (mrakitin) [inline setup.py]:
```suggestion
    extra_link_args = ['-shared']
```

File: setup.py [hunk 2]
```diff
@@ -35,8 +35,13 @@
...
+else:
+    extra_link_args=[]
```
Comment (mrakitin) [inline setup.py]:
```suggestion
    extra_link_args = []
```

**Comment by shenmidelin** [comment]
I have tagged it as v0.0.4, and upload it to pypi. We can now work in the source from PyPi.

**Comment by mrakitin** [comment]
Thank you! I will give my feedback on https://github.com/mrakitin/lightsource2-recipes/pull/1 later today.

**Pull request merged**